data.go

```diff
@@ -174,49 +183,70 @@ func (d *Data) Datasource(alias string, args ...string) map[string]interface{} {
 	if err != nil {
 		log.Fatalf("Couldn't read datasource '%s': %s", alias, err)
 	}
-	if source.Type == "application/json" {
-		ty := &TypeConv{}
-		return ty.JSON(string(b))
-	}
-	if source.Type == "application/yaml" {
-		ty := &TypeConv{}
-		return ty.YAML(string(b))
-	}
-	log.Fatalf("Datasources of type %s not yet supported", source.Type)
-	return nil
+	return b
 }
 
 // ReadSource -
-func (d *Data) ReadSource(fs vfs.Filesystem, source *Source, args ...string) ([]byte, error) {
-	if d.cache == nil {
-		d.cache = make(map[string][]byte)
-	}
-	cacheKey := source.Alias
-	for _, v := range args {
-		cacheKey += v
-	}
-	cached, ok := d.cache[cacheKey]
-	if ok {
+func (d *Data) ReadSource(fs vfs.Filesystem, source *Source, args ...string) (map[string]interface{}, error) {
```

Comment:
I changed it since it was necessary to add the creation of the JSON and YAML documents in the mutex protected section when doing parallel processing. Also, I think it makes sense to include the (potentially costly) deserialized objects into the cache so that this conversion needs to be called only once (and not for every call to the datasource). I don't consider this optimisation as premature since the call to the datasource can happen quite often also when doing only a single call.

I suggest adding a second method `ReadSourceAsObjects` (could be private, too) which is used by DataSource and leave an (unoptimised) `ReadSource` with the original signature. Then we could decide whether to increase the cache to hold both kinds of objects (with the cost of increasing memory usage) or we make `ReadSource` to not use a cache.

wdyt ?